include/net/xia_ether.h

```diff
@@ -0,0 +1,347 @@
+#ifndef _NET_XIA_ETHER_H
+#define _NET_XIA_ETHER_H
+/* prevents double declarations */
+
+#include <linux/netdevice.h>
+#include <linux/etherdevice.h>
+#include <linux/err.h>
+#include <net/xia_list_fib.h>
+#include <linux/netlink.h>
+#include <linux/spinlock.h>
+
+/* Ethernet Principal */
+#define XIDTYPE_ETHER (__cpu_to_be32(0x22))
+
+struct rtnl_xia_ether_addrs {
+	__u16		attr_len;
+	__u8		interface_addr_len;
+	__u8		interface_addr[MAX_ADDR_LEN];
+	int		interface_index;
+};
+
+static inline int RTHA_OK(struct rtnl_xia_ether_addrs *rtha, int len)
+{
+	return len >= 0 && (unsigned)len >= sizeof(*rtha) &&
+		rtha->attr_len <= (unsigned)len;
+}
+
+static inline struct rtnl_xia_ether_addrs *RTHA_NEXT(
+	struct rtnl_xia_ether_addrs *rtha)
+{
+	return	(struct rtnl_xia_ether_addrs *)
+		(((char *)rtha) + NLMSG_ALIGN(rtha->attr_len));
+}
+
+#ifdef __KERNEL__
+/* only for kernel use */
+
+/* ETHER's virtal XID type. */
+int ether_vxt __read_mostly = -1;
+
+/* Local ETHERs */
+
+struct fib_xid_ether_local {
+	struct xip_dst_anchor	xel_anchor;
+
+	/* WARNING: @xel_common is of variable size, and
+	 * MUST be the last member of the struct.
+	 */
+	struct fib_xid		xel_common;
+};
+
+static inline struct fib_xid_ether_local *fxid_lether(struct fib_xid *fxid)
+{
+	return likely(fxid)
+		? container_of(fxid, struct fib_xid_ether_local, xel_common)
+		: NULL;
+}
+
+/* Main ETHERs */
+struct ether_interface{
```

Comment:
Please add a space between the struct definition and ``{``, i.e., ``struct ether_interface {``.